demos/jobs/jobs_demo_mosquitto/jobs_demo_mosquitto.c

```diff
@@ -0,0 +1,1264 @@
+             "  --document '{\"url\":\"https://cdn.kernel.org/pub/linux/kernel/v5.x/linux-5.8.5.tar.xz\"}'\n"
+             "\nTo execute the job, on the target device run the downloader with the target's credentials, e.g.,\n"
+             "$ %s -n device1 -h abcdefg123.iot.us-east-1.amazonaws.com \\\n"
+             "  --cert bbaf123456-certificate.pem.crt --key bbaf123456-private.pem.key\n"
+             "\nTo exit the program, type Control-C, or send a SIGTERM signal.\n",
+             programName );
+    fprintf( stderr,
+             "\nOutput should look like the following:\n"
+             "Connecting to abcdefg123.iot.us-east-1.amazonaws.com, port 8883.\n"
+             "Client device1 sending CONNECT\n"
+             "Client device1 received CONNACK\n"
+             "Client device1 sending SUBSCRIBE (Mid: 1, Topic: $aws/things/device1/jobs/start-next/accepted, QoS: 1)\n"
+             "Client device1 received SUBACK\n"
+             "[...]\n"
+             "starting job id: t12\n"
+             "sending first update\n" );
+    fprintf( stderr,
+             "\nIf the output does not show a successful connection, check in the AWS console\n"
+             "that the client certificate is associated with the target thing and is activated.\n"
+             "Also check that the Amazon Root CA certificates are in your system's trust store.\n"
+             "Note, you can provide a CA certificate file directly as a command-line argument.\n" );
+    fprintf( stderr,
+             "\nThis demonstration exits on most error conditions.  One way to retry while avoiding\n"
+             "throttling due to excessive reconnects is to periodically relaunch from cron(8).\n"
+             "Given a shell script wrapper with the necessary arguments called download, the following\n"
+             "line in /etc/crontab would start the downloader unless it is already running.\n"
+             "This tries every 3 minutes, with an additional random delay up to 2 minutes.\n\n"
+             "*/3 * * * *  root  exec 9> /tmp/lock && flock -n 9 && sleep $((RANDOM %% 120)) && download\n"
+             );
+    fprintf( stderr,
+             "\nusage: %s "
+             "-n name -h host [-p port] {--cafile file | --capath dir} --cert file --key file\n"
+             "\n"
+             "-n : thing name\n"
+             "-h : mqtt host to connect to.\n"
+             "-p : network port to connect to. Defaults to %d.\n",
+             programName, DEFAULT_MQTT_PORT );
+    fprintf( stderr,
+             "--cafile : path to a file containing trusted CA certificates to enable encrypted\n"
+             "           certificate based communication.\n"
+             "--capath : path to a directory containing trusted CA certificates to enable encrypted\n"
+             "           communication.  Defaults to %s.\n"
+             "--cert : client certificate for authentication.\n"
+             "--key : client private key for authentication.\n\n",
```

Comment:
I feel like from reading `--cert` and `--key` should be labeled as `--certfile` and `--keyfile` in. Also, it is not immediately obvious to me whether they should be in PEM or DER format, so I feel that must be explicitly stated.

Reply:
added